e2e test failing: tests/test_e2e_05_topology.py::TestE2ETopology::test_515_delete_interface_automatically

It was merged yesterday on PR #297:

The code looks correct, it also performs a sleep and wait for switches to connect before trying to assert, needs some investigation cc'ing @Alopalao:

```
tests/test_e2e_05_topology.py::TestE2ETopology::test_515_delete_interface_automatically: 2024-04-17,04:28:32.219585 - 2024-04-17,04:28:32.228130
self = <tests.test_e2e_05_topology.TestE2ETopology object at 0x7f8ccf759ed0>
    def test_515_delete_interface_automatically(self):
        """Test interface removal after logical deletion"""
        intf_id = "00:00:00:00:00:00:00:01:4"
        api_url = KYTOS_API + f'/topology/v3/interfaces'
        response = requests.get(api_url)
        assert response.status_code == 200, response.text
        data = response.json()
>       assert intf_id in data["interfaces"]
E       AssertionError: assert '00:00:00:00:00:00:00:01:4' in {'00:00:00:00:00:00:00:01:1': {'active': True, 'enabled': False, 'id': '00:00:00:00:00:00:00:01:1', 'link': '', ...}, ...0:01:4294967294': {'active': True, 'enabled': False, 'id': '00:00:00:00:00:00:00:01:4294967294', 'link': '', ...}, ...}
tests/test_e2e_05_topology.py:930: AssertionError
tests/test_e2e_05_topology.py::TestE2ETopology::test_515_delete_interface_automatically: 2024-04-17,04:28:56.683765 - 2024-04-17,04:28:56.692564
=========================== rerun test summary info ============================
RERUN tests/test_e2e_05_topology.py::TestE2ETopology::test_515_delete_interface_automatically
RERUN tests/test_e2e_05_topology.py::TestE2ETopology::test_515_delete_interface_automatically
=========================== short test summary info ============================
FAILED tests/test_e2e_05_topology.py::TestE2ETopology::test_515_delete_interface_automatically - AssertionError: assert '00:00:00:00:00:00:00:01:4' in {'00:00:00:00:00:00:00:01:1': {'active': True, 'enabled': False, 'id': '00:00:00:00:00:00:00:01:1', 'link': '', ...}, ...0:01:4294967294': {'active': True, 'enabled': False, 'id': '00:00:00:00:00:00:00:01:4294967294', 'link': '', ...}, ...}
= 1 failed, 238 passed, 8 skipped, 9 xfailed, 7 xpassed, 1143 warnings, 2 rerun in 12275.20s (3:24:35) =
```

--- a/tests/test_e2e_05_topology.py
+++ b/tests/test_e2e_05_topology.py
@@ -919,22 +919,3 @@ class TestE2ETopology:
         assert response.status_code == 200, response.text
         data = response.json()
         assert not intf_id in data["interfaces"]
-
-    def test_515_delete_interface_automatically(self):
-        """Test interface removal after logical deletion"""
-        intf_id = "00:00:00:00:00:00:00:01:4"
-        api_url = KYTOS_API + f'/topology/v3/interfaces'
-        response = requests.get(api_url)
-        assert response.status_code == 200, response.text
-        data = response.json()
-        assert intf_id in data["interfaces"]
-
-        s1 = self.net.net.get('s1')
-        s1.detach('s1-eth4')
-        time.sleep(5)
-
-        api_url = KYTOS_API + f'/topology/v3/interfaces'
-        response = requests.get(api_url)
-        assert response.status_code == 200, response.text
-        data = response.json()
-        assert not intf_id in data["interfaces"]

--- a/tests/test_e2e_06_topology.py
+++ b/tests/test_e2e_06_topology.py
@@ -1,0 +1,49 @@
+import requests
+from tests.helpers import NetworkTest
+import time
+
+CONTROLLER = '127.0.0.1'
+KYTOS_API = 'http://%s:8181/api/kytos' % CONTROLLER
+
+class TestE2ETopology:
+    net = None
+
+    def setup_method(self, method):
+        """
+        It is called at the beginning of every class method execution
+        """
+        # Start the controller setting an environment in
+        # which all elements are disabled in a clean setting
+        self.net.start_controller(clean_config=True, enable_all=False)
+        self.net.wait_switches_connect()
+        time.sleep(10)
+
+    @classmethod
+    def setup_class(cls):
+        cls.net = NetworkTest(CONTROLLER)
+        cls.net.start()
+        cls.net.wait_switches_connect()
+        time.sleep(10)
+
+    @classmethod
+    def teardown_class(cls):
+        cls.net.stop()
+
+    def test_010_delete_interface_automatically(self):
+        """Test interface removal after logical deletion"""
+        intf_id = "00:00:00:00:00:00:00:02:1"
+        api_url = KYTOS_API + f'/topology/v3/interfaces'
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert intf_id in data["interfaces"]
+
+        s2 = self.net.net.get('s2')
+        s2.detach('s2-eth1')
+        time.sleep(5)
+
+        api_url = KYTOS_API + f'/topology/v3/interfaces'
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert not intf_id in data["interfaces"]

